tests/lib/tools/log-analyzer

```diff
@@ -0,0 +1,511 @@
+        elif results == NONE:
+            pass
+        elif results == FAILED:
+            print(''.join(str(x) for x in self.full_log if x.type == RESULT and \
+                x.result_type == FAILED_TYPE))
+        elif results == ABORTED:
+            print(''.join(str(x) for x in self.full_log if x.type == RESULT and \
+                x.result_type == ABORTED_TYPE))
+        else:
+            print(''.join(str(x) for x in self.full_log if x.type == RESULT and \
+                x.result_type == SUCCESSFUL_TYPE))
+
+
+    def export_log(self, filepath):
+        prepared_log = []
+        for item in self.full_log:
+            prepared_log.append(item.__dict__())
+        with open(filepath, 'w') as json_file:
+            json.dump(prepared_log, json_file)
+
+    def _next_line(self):
+        self.iter = self.iter + 1
+        return self.lines[self.iter-1]
+
+    def check_log_exists(self):
+        return os.path.exists(self.filepath)
+
+    def read_spread_log(self):
+        with open(self.filepath) as filepath:
+            self.lines = filepath.readlines()
+            self.iter = 0
+
+            if self.store_setup:
+                while self.iter < len(self.lines):
+                    line = self._next_line()
+                    if self._match_start(line):
+                        break
+                    self.full_log.append(line)
+
+                if self.iter >= len(self.lines):
+                    # Start not found, the log could be either empty, corrupted or cut
+                    self.iter = 0
+
+            while self.iter < len(self.lines):
+                line = self._next_line()
+                if self._match_task(line):
+                    action = self._get_action(line)
+                    if action:
+                        #print(action)
+                        self.full_log.append(action)
+                    continue
+
+                if self._match_info(line):
+                    info = self._get_info(line)
+                    if info:
+                        #print(info)
+                        self.full_log.append(info)
+                    continue
+
+                if self._match_operation(line):
+                    operation = self._get_operation(line)
+                    if operation:
+                        #print(operation)
+                        self.full_log.append(operation)
+                    continue
+
+                if self._match_result(line):
+                    result = self._get_result(line)
+                    if result:
+                        #print(result)
+                        self.full_log.append(result)
+                    continue
+
+    def _match_date(self, date):
+        return re.findall(r'\d{4}-\d{2}-\d{2}', date)
+
+    def _match_time(self, time):
+        return re.findall(r'\d{2}:\d{2}:\d{2}', time)
+
+    def _match_info(self, line):
+        parts = line.split(' ')
+        return len(parts) > 3 and \
```

Comment:
I would suggest running `black` on the source file, as IIRC long lines are now customarily wrapped by placing them in parenthesis.